### PR TITLE
feat: add variables outputs

### PR DESCRIPTION
**Pull request by luizstacio**
### Summary

- Add `variableOutputs` to contract call overrides `Overrides`.  This adds an `x` number of variable outputs as determined 

### Example
```ts
await token.functions.transfer_coins_to_output(amount, tokenId, address, {
    variableOutputs: 1,
});
```

closes: #223

### CODE REVIEW

**Review by QuinnLee — APPROVED** [review]
👍  most of the code seems to be sway and binary files.  The TS looks good

File: packages/contract/scripts/build-contracts.sh [hunk 1]
```diff
@@ -1,2 +1,5 @@
+#!/bin/sh
+
 forc build -p src/storage-test-contract --print-finalized-asm
-forc build -p src/call-test-contract --print-finalized-asm
\ No newline at end of file
+forc build -p src/call-test-contract --print-finalized-asm
+forc build -p src/token-test-contract/token_contract --print-finalized-asm
```
Comment (adlerjohn) [inline packages/contract/scripts/build-contracts.sh:5]: Do you actually need to print asm here?

File: packages/contract/src/token-test-contract/token_contract/Forc.toml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[project]
+authors = ["Mohammad"]
```
Comment (adlerjohn) [inline packages/contract/src/token-test-contract/token_contract/Forc.toml:2]:
```suggestion
authors = ["Fuel Labs <contact@fuel.sh>"]
```

**Review by QuinnLee — APPROVED** [review]

**Review by AlicanC — APPROVED** [review]

File: packages/contract/src/contract.ts [hunk 3]
```diff
@@ -32,6 +33,29 @@ const getOverrides = (func: FunctionFragment, args: Array<any>) => {
...
 
+export const buildTransaction = (
+  contract: Contract,
+  func: FunctionFragment,
+  args: Array<any>
+): ScriptTransactionRequest => {
+  const overrides = getOverrides(func, args);
+  const data = contract.interface.encodeFunctionData(func, args);
+  const request = new ScriptTransactionRequest({
+    gasLimit: 1000000,
```
[the remark this answers is not in the document]
Comment (adlerjohn) [reply packages/contract/src/contract.ts:44]: (can be done in a future PR, just file an issue)

[the remark this answers is not in the document]
Comment (luizstacio) [reply packages/contract/src/contract.ts:44]: https://github.com/FuelLabs/fuels-ts/issues/225

**Pull request merged**